crates/bevy_input/src/common_conditions.rs

```diff
@@ -0,0 +1,97 @@
+use crate::Input;
+use bevy_ecs::system::Res;
+use std::hash::Hash;
+
+/// Stateful run condition that can be toggled via a input press using [`Input::just_pressed`].
+///
+/// ```rust,no_run
+/// use bevy::prelude::*;
+/// use bevy::input::common_conditions::input_toggle_active;
+///
+/// fn main() {
+///     App::new()
+///         .add_plugins(DefaultPlugins)
+///         .add_system(pause_menu.run_if(input_toggle_active(false, KeyCode::Escape)))
+///         .run();
+/// }
+///
+/// fn pause_menu() {
+///     println!("in pause menu");
+/// }
+/// ```
+///
+/// If you want other systems to be able to access whether the toggled state is active,
+/// you should use a custom resource or a state for that:
+/// ```rust,no_run
+/// use bevy::prelude::*;
+/// use bevy::input::common_conditions::input_toggle_active;
+///
+/// #[derive(Resource, Default)]
+/// struct Paused(bool);
+///
+/// fn main() {
+///     App::new()
+///         .add_plugins(DefaultPlugins)
+///         .init_resource::<Paused>()
+///         .add_system(pause_menu.run_if(|paused: Res<Paused>| paused.0))
+///         .run();
+/// }
+///
+/// fn update_pause_state(mut paused: ResMut<Paused>, input: Input<KeyCode>) {
+///   if input.just_pressed(KeyCode::Escape) {
+///     paused.0 = !paused.0;
+///   }
+/// }
+///
+/// fn pause_menu() {
+///     println!("in pause menu");
+/// }
+///
+/// ```
+pub fn input_toggle_active<T>(default: bool, input: T) -> impl FnMut(Res<Input<T>>) -> bool
+where
+    T: Copy + Eq + Hash + Send + Sync + 'static,
+{
+    let mut active = default;
+    move |inputs: Res<Input<T>>| {
+        active ^= inputs.just_pressed(input);
```

Comment:
Would you mind explaining what this symbol is `^=` for my smooth brain pls lol. I think I understand but it'd be worth clarifying haha.

Reply:
Yes, okay I think that makes sense to me lol.